src/main/scala/uk/gov/nationalarchives/tdr/api/auth/AuthorisationTags.scala

```diff
@@ -182,20 +181,17 @@ case class ValidateUserOwnsFiles[T](argument: Argument[T]) extends Authorisation
       throw InputDataException(s"'fileIds' is empty. Please provide at least one fileId.")
     }
     for {
-      fileIdsAndOwner: Seq[FileOwnership] <- ctx.ctx.fileService.getOwnersOfFiles(fileIds)
-      fileIdsBelongingToAConsignment: Seq[UUID] = fileIdsAndOwner.map(_.fileId)
-      filesThatDoNotBelongToAConsignment: Seq[UUID] = fileIds.filterNot(fileId => fileIdsBelongingToAConsignment.contains(fileId))
-      fileIdsThatDoNotBelongToTheUser: Seq[UUID] = fileIdsAndOwner.collect {
-        case fileIdAndOwner if fileIdAndOwner.userId != userId => fileIdAndOwner.fileId
+      fileOwner: Seq[(UUID, UUID)] <- ctx.ctx.fileService.getOwnersOfFiles(fileIds)
+      fileIdsThatDoNotBelongToTheUser: Seq[UUID] = fileOwner.collect {
+        case (fileId, ownerId) if ownerId != userId => fileId
       }
-      allFilesBelongToAConsignment = filesThatDoNotBelongToAConsignment.isEmpty
-      allFilesBelongToTheUser = fileIdsThatDoNotBelongToTheUser.isEmpty
+      fileIdsWithNoOwner: Seq[UUID] = fileIds.filterNot(fileId => fileOwner.exists(_._1 == fileId))
```

Comment:
Looking the logic here again, it should be possible to simplify it a bit if you combine the `fileIdsThatDoNotBelongToTheUser` and `fileIdsWithNoOwner` into a single list.

This single list can then be used in the `if` statement, ie `if (list.isEmpty || exportAccess)`.

That single list can then be used in the error message, which will then mean the json responses will be correct, without the need to change them.